crates/tabby/src/services/completion.rs

```diff
@@ -118,6 +118,9 @@ pub struct Segments {
     /// or the git repository does not have a remote URL.
     git_url: Option<String>,
 
+    /// The code snippets that are related for code completion.
+    snippets: Option<Vec<Snippet>>,
+
```

Comment:
```suggestion
    /// TODO(zhiming): explain how the declarations are extracted briefly
    relevant_declaration: Vec<Declaration>,
```

Always name the field precisely.